### ISSUE TITLE
[NTI] Support getters/setters as first-class aspects of Property

### ISSUE THREAD

**Issue by shicks** [google/closure-compiler#2545]
Currently NTI represents getters by storing a normal `Property` object under a different key.  For example,

```js
const x = {foo: 5};
const y = {get foo() { return 5; }};
```

In this case, the property map for `x` is `{foo: number}`, while the property map for `y` is `{%getter_deffoo: number}`.  This is a hack and means that every time a property look-up happens, we need to add additional logic that understands how to add one or the other (getter/setter) prefix, depending on context that may not even be known.

It would be more correct to teach the `Property` class how to represent getters and setters, and then to store the property under the correct name.

**Comment by EatingW** [comment]
NTI tests seems to generate wrong warnings. Currently NTI expects 
```
var x = { get a() { return 1; } }; x.a = 'str';
```
And complain about illegal property creation on:
```
var x = /** @struct */ { get a() {} }; x.a = 123;
```
This seems to be due to the inconsistent way it is treating getter properties. 
If we change the mayHaveProp method to specifically look for getters, then it does not produce the warning in the second case. However, in JavaScript, in strict mode both cases has an error. In sloppy mode it silently does nothing.
